src/dasi/api/ObjectLocation.h

```diff
@@ -0,0 +1,42 @@
+
+#pragma once
+
+#include <string>
+
+namespace dasi::api {
+
+class ReadHandle;
+
+//----------------------------------------------------------------------------------------------------------------------
+
+class ObjectLocation {
+
+public: // types
+
+    using offset_type = long long;
+    using length_type = unsigned long long;
+
+public: // methods
+
+    ObjectLocation(const std::string& type, const std::string& location, offset_type offset, length_type length);
```

Comment:
This is interesting. I would probably err towards having a factory of ObjectLocation types (which then directly know how to construct their own read handle) rather than just storing a string and building the ReadHandle from a factory.

That said I don't feel strongly about it, and would happily hear an opposing argument?

Reply:
I don't feel strongly about it either, the main reasons why I chose to have a single ObjectLocation type are that I felt we didn't need to have parallel flavours of both ObjectLocation and ReadHandle, and because it made it easier to have serialisation and deserialisation in the same place. If we go for a factory for the ObjectLocations, that means we will have to manage pointers again. It's not that much of an issue since there is no underlying handle to be closed, but still slightly cumbersome for limited benefit. Happy to discuss.

Reply:
That's reasonable. Can we make a note in the code explicitly though that if this becomes constraining, then this is the place to add extra functionality.